### PR TITLE
replace patchesStrategicMerge with patches

### LINKED ISSUE #1564: Replace `patchesStrategicMerge`/`patchesJson6902` with `patches`
### Describe request or inquiry 
* [Kustomize 5.0.0](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0) was released Feb 2023. The release notes mention:
> We are deprecating the following kustomization fields. These fields will never be removed from the kustomize.config.k8s.io/v1beta1 Kustomization API, but they will not be included in the kustomize.config.k8s.io/v1 Kustomization API. When Kustomization v1 is available, we will announce the deprecation of the v1beta1 version. There will be at least two releases between deprecation and removal of Kustomization v1beta1 support from the kustomize CLI, and removal itself will happen in a future major version bump.
> * patchesStrategicMerge. Please use [patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/) instead
> * patchesJson6902. Please use [patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/) instead
> * vars. Please use [replacements](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/) instead.
> * imageTags. This is an alias for [images](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/images/), please switch to images.
* Today, Online Boutique uses `patchesStrategicMerge` and `patchesJson6902`. We should replace the use of these fields with [`patches`](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/).

### What purpose/environment will this feature serve? 
* This impacts a few of [our Kustomize components](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/kustomize).

### PR DESCRIPTION
**Pull request by smeet07**
### Fixes 
#1564

### CODE REVIEW

**Comment by smeet07** [comment]
R: @NimJay

**Comment by bourgeoisor** [comment]
This pipeline is failing: https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/.github/workflows/kustomize-build-ci.yaml

@smeet07 could look into this + add some more context in the description?

**Comment by smeet07** [comment]
> This pipeline is failing: https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/.github/workflows/kustomize-build-ci.yaml
> 
> @smeet07 could look into this + add some more context in the description?

The patchesStrategicMerge has been deprecated and hence needs to be replaced by patches

**Comment by smeet07** [comment]
@NimJay

File: kustomize/components/cymbal-branding/kustomization.yaml [hunk 1]
```diff
@@ -14,18 +14,18 @@
...
-    template:
-      spec:
-        containers:
-          - name: server
-            env:
-            - name: CYMBAL_BRANDING
-              value: "true"
+patches:
+  # frontend - enable CYMBAL_BRANDING
+  - |-
```
Comment (minherz) [inline kustomize/components/cymbal-branding/kustomization.yaml:19]: there is extra indentation that breaks the format. Please shift left _all_ content under `patches`

**Review by minherz — CHANGES_REQUESTED** [review]

**Comment by minherz** [comment]
@NimJay , you probably will need to fix the problem with the required code-tests and deployment-tests checks using the [recommended technique][1].

[1]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks

**Review by minherz — COMMENTED** [review]
the syntax for `patches` is incorrect. The problem is not only in the cymbal-branding component but in all components that have inline patch.
See [examples] for inline patching. TL;DR; you need to define a field `patch:` for each element of the `patches` list.

I fixed it for you.

[examples]: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/inlinePatch.md

**Pull request merged**